### ISSUE TITLE
Remove security warning from old lodash

### ISSUE THREAD

**Issue by mroderick** [sinonjs/sinon#1948]
When running `npm audit`, there is a warning about `lodash`.

1. Run `npm audit`
1. Observe the warning about lodash (example below)

```
┌───────────────┬────────────────────────────────────────┐
│ Low           │ Prototype Pollution                    │
├───────────────┼────────────────────────────────────────┤
│ Package       │ lodash                                 │
├───────────────┼────────────────────────────────────────┤
│ Patched in    │ >=4.17.5                               │
├───────────────┼────────────────────────────────────────┤
│ Dependency of │ mochify-istanbul [dev]                 │
├───────────────┼────────────────────────────────────────┤
│ Path          │ mochify-istanbul > lodash              │
├───────────────┼────────────────────────────────────────┤
│ More info     │ https://nodesecurity.io/advisories/577 │
└───────────────┴────────────────────────────────────────┘
```

The old `lodash` that has the issue  is used by `mochify-istanbul`, which is part of the `test-coverage`task.

I've tried [upgrading `lodash` in `mochify-istanbul`](https://github.com/ferlores/mochify-istanbul), but that was not successful. To me that project looks pretty dead.

We should consider other solutions for coverage. I have a couple of ideas:

1. use `nyc` to instrument the files for mochify (this might take some work because of `proxyquire-universal`)
1. use `jsdom` to fill in the missing apis, so all tests can be run in node ... and then use `nyc` to instrument files for mocha

Other ideas welcome

**Comment by mantoni** [comment]
The project isn't dead. It worked just fine all the time. I'm happy to resurrect it.

**Comment by mantoni** [comment]
Ok, I had a quick look at `mochify-istanbul`, and although it works, the tests don't and the whole project is in a terrible state. The owner seems to have abandoned GitHub entirely.

The way forward seems indeed to use `nyc`. There is a babel plugin for instrumentation that could be used. I have good experience with using `babelify` to get any babel plugin to work with the browserify pipeline. @mroderick You know `nyc` better than me. Is there an easier way?

**Comment by mroderick** [comment]
I did some experiments yesterday, and couldn't get `nyc` working with `mochify` in the first try.

I don't know if there's an easy way out of this. Option 1 requires studying how all the pipelining works, option 2 requires changing some of the tests. I do see option two to have the added benefit of being able to easily use the debugger with node for those tests also.

Any `nyc` experts out there who can offer advice?

**Comment by mantoni** [comment]
Fixed by #1950.

**Comment by mroderick** [comment]
:
